test/integration/integration.cc

```diff
@@ -765,81 +777,83 @@ void BaseIntegrationTest::testTwoRequests(Http::CodecClient::Type type) {
   Buffer::OwnedImpl buffer("GET / HTTP/1.1\r\n\r\n");
   std::string response;
   RawConnectionDriver connection(
-      lookupPort("http"), buffer,
-      [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
+      lookupPort("http"),
+      buffer, [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
         response.append(TestUtility::bufferToString(data));
         client.close(Network::ConnectionCloseType::NoFlush);
-      });
+      }, version);
 
   connection.run();
   EXPECT_TRUE(response.find("HTTP/1.1 400 Bad Request\r\n") == 0);
 }
 
-void BaseIntegrationTest::testBadPath() {
+void BaseIntegrationTest::testBadPath(Network::Address::IpVersion version) {
   Buffer::OwnedImpl buffer("GET http://api.lyft.com HTTP/1.1\r\nHost: host\r\n\r\n");
   std::string response;
   RawConnectionDriver connection(
-      lookupPort("http"), buffer,
-      [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
+      lookupPort("http"),
+      buffer, [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
         response.append(TestUtility::bufferToString(data));
         client.close(Network::ConnectionCloseType::NoFlush);
-      });
+      }, version);
 
   connection.run();
   EXPECT_TRUE(response.find("HTTP/1.1 404 Not Found\r\n") == 0);
 }
 
-void BaseIntegrationTest::testUpstreamProtocolError() {
+void BaseIntegrationTest::testUpstreamProtocolError(Network::Address::IpVersion version) {
   IntegrationCodecClientPtr codec_client;
   IntegrationStreamDecoderPtr response(new IntegrationStreamDecoder(*dispatcher_));
   FakeRawConnectionPtr fake_upstream_connection;
-  executeActions(
-      {[&]() -> void {
-        codec_client = makeHttpConnection(lookupPort("http"), Http::CodecClient::Type::HTTP1);
-      },
-       [&]() -> void {
-         codec_client->startRequest(Http::TestHeaderMapImpl{{":method", "GET"},
-                                                            {":path", "/test/long/url"},
-                                                            {":authority", "host"}},
-                                    *response);
-       },
-       [&]() -> void { fake_upstream_connection = fake_upstreams_[0]->waitForRawConnection(); },
-       // TODO(mattklein123): Waiting for exact amount of data is a hack. This needs to
-       // be fixed.
-       [&]() -> void { fake_upstream_connection->waitForData(187); },
-       [&]() -> void { fake_upstream_connection->write("bad protocol data!"); },
-       [&]() -> void { fake_upstream_connection->waitForDisconnect(); },
-       [&]() -> void { codec_client->waitForDisconnect(); }});
+  executeActions({[&]() -> void {
+    codec_client = makeHttpConnection(lookupPort("http"), Http::CodecClient::Type::HTTP1, version);
+  },
+                  [&]() -> void {
+                    codec_client->startRequest(Http::TestHeaderMapImpl{{":method", "GET"},
```

Comment:
Formatting here looks weird, the first block is done one way, the next another...